schema/bom-1.6.schema.json

```diff
@@ -1759,6 +1765,12 @@
           },
           "title": "Depends On",
           "description": "The bom-ref identifiers of the components or services that are dependencies of this dependency object."
+        },
+        "type": {
```

Comment:
❌ i'd veto this.

this is a bad idea of building a relation between "components" and "interfaces/algorithms to implement".
reason: if a component A depends on component B and A also implements C, then i would have to double A in the dependency graph, like 
```json
[
  { "ref": "A", "depedsOn": ["B"], "type": "use" },
  { "ref": "A", "depedsOn": ["C"], "type": "implements" } 
]
```




Alternative to consider:
- alongside to easing `dependsOn` - add a `provides`, which would be a list of algorithms/interfaces a component (determined by `ref`) actually fulfills.
  example:
  ```json
  [
    { "ref": "A", "depedsOn": ["B"], "privides": "C" }
  ]
  ```

Reply:
So for this suggestion, this would change to the following:
```json
{
  "bomFormat": "CycloneDX",
  "specVersion": "1.6",
  "serialNumber": "urn:uuid:3e671687-395b-41f5-a30f-a58921a69b79",
  "version": 1,
  "metadata": {
    "component": {
      "type": "application",
      "bom-ref": "acme-application",
      "name": "Acme Application",
      "version": "1.0"
    }
  },
  "components": [
    {
      "type": "cryptographic-asset",
      "bom-ref": "aes128gcm",
      "name": "AES",
      "cryptoProperties": {
        "assetType": "algorithm",
        "algorithmProperties": {
          "primitive": "ae",
          "parameterSetIdentifier": "128",
          "executionEnvironment": "software-plain-ram",
          "implementationPlatform": "x86_64",
          "certificationLevel": [ "none" ],
          "mode": "gcm",
          "cryptoFunctions": ["keygen", "encrypt", "decrypt", "tag"],
          "classicalSecurityLevel": 128,
          "nistQuantumSecurityLevel": 1
        },
        "oid": "oid:2.16.840.1.101.3.4.1.6"
      }
    },
    {
      "type": "library",
      "bom-ref": "crypto-library",
      "name": "Crypto library",
      "version": "1.0.0"
    }
  ],
  "dependencies": [
    {
      "ref": "acme-application",
      "dependsOn": ["crypto-library"]
    },
    {
      "ref": "crypto-library",
      "provides": ["aes128gcm"]
    }
  ]
}
```

Is this what you're thinking?

Reply:
> ```json
> [
>   { "ref": "A", "depedsOn": ["B"], "provides": ["C"] }
> ]
> ```

So dependsOn implies "uses" and provides implies "implements", correct?  If so that's fine with me.

Reply:
TODO for @jkowalleck 
- add a definition for "provides" and what it makes it different from "dependsOn". 
  be precise wit the definition. see to reuse the existing one from "implements".
  Drop a PR pointing to `1.6-dev-cbom`

Reply:
@stevespringett @bhess @prabhu
change request ready for review: https://github.com/CycloneDX/specification/pull/366